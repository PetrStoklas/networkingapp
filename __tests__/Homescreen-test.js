/**
 * @format
 */

import 'react-native';
import React from 'react';
import HomeScreen from '../src/screens/HomeScreen';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';

const sleep = (ms) => new Promise((resolve) => setTimeout(() => resolve(true), ms));

jest.mock('../src/hooks/useAuth', () => {
  return {
    ...jest.requireActual('../src/hooks/useAuth'),
    useAuth: jest.fn(() => {
      return {
        user: {
          photoUrl: 'asd',
        },
        profile: {
          name: 'rofl',
        },
      };
    }),
  };
});

it('renders correctly', async () => {
  const app = renderer
    .create(
      <NavigationContainer>
        <HomeScreen />
      </NavigationContainer>
    )
    .toJSON();
  expect(app).toMatchSnapshot();
  await sleep(10);
});
