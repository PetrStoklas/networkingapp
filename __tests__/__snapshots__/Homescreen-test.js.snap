// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={Object {}}
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={Object {}}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 38,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï‡¸
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <Text>
      This is homescreen
    </Text>
    <Text>
      Welcome 
      rofl
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign out
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
